src/components/Button/Button.jsx

```diff
@@ -39,96 +41,110 @@ const Button = forwardRef(({ children, icon, direction, caption, captionColor, d
   </Base>
 ))
 
-const colorVariants = variant({
-  default: 'filled',
-  key: 'button',
-  variants: {
-    filled: css`
-      background-color: ${({ color }) => th.color(color)};
-      border-color: ${({ color }) => th.color(color)};
-      color: white;
-      &:hover {
-        background-color: ${({ color }) => th.color(`${color}_hover`)};
-        border-color: ${({ color }) => th.color(`${color}_hover`)};
+const colorVariants = props => {
+  return variant({
+    prop: 'variant',
+    variants: {
+      filled: {
+        backgroundColor: `${props.theme.colors[`${props.color}`]}`,
+        borderColor: `${props.theme.colors[`${props.color}`]}`,
+        color: 'white'
+      },
+      outlined: {
+        backgroundColor: 'transparent',
+        borderColor: `${props.theme.colors[`${props.color}`]}`,
+        color: `${props.theme.colors[`${props.color}`]}`
+      },
+      text: {
+        backgroundColor: 'transparent',
+        borderColor: 'transparent',
+        color: `${props.theme.colors[`${props.color}`]}`
       }
-      &:active {
-        background-color: ${({ color }) => th.color(`${color}_active`)};
-        border-color: ${({ color }) => th.color(`${color}_active`)};
-      }
-      &:disabled {
-        background-color: disabled;
-        border-color: disabled;
+    }
+  })
+}
+
+const StyledIcon = styled(Icon)`
+  ${colorVariants}
+`
+
+const Base = styled.button(
+  props => css`
```

Comment:
Se aquilo de desestruturar o theme for uma boa, da para implementar aqui também

Reply:
Daria para desestruturar o theme , variant e o color

Reply:
Assim como o Carrilhos falou, dá pra desestruturar desta forma e utilizar assim:
```
const colorVariants = ({ theme: { colors }, color }) => {
return variant({
    prop: 'variant',
    variants: {
      filled: {
        backgroundColor: `${colors[`${color}`]}`,
        ...
```